Commit fixed indexing load and parallel query challenge

dliappis opened the pull request:
Add a challenge that runs three parallel queries while running in
parallel an indexing task with fixed throughput (supplied via a track
param).

Also fix track parameter bugs in README and allow specifying the
index_prefix in the challenges `index-and-query-logs-fixed-daily-volume`
and `index-logs-fixed-daily-volume`.

dliappis commented:
@danielmitterdorfer hold of a review, I caught a bug with the `index-logs-fixed-daily-volume` in IT tests. Will ping you when ready.

dliappis commented:
@danielmitterdorfer Fixed, you can proceed with the review whenever you want.

danielmitterdorfer commented on eventdata/challenges/daily-log-index-fixed-throughput-and-query.json:
```diff
@@ -0,0 +1,137 @@
+  "schedule": [
+    {
+      "name": "check-cluster-health",
+      "operation": {
```
we should add a `"retry-until-success": true` here (similar to https://github.com/elastic/rally-eventdata-track/pull/97).

danielmitterdorfer commented on eventdata/challenges/daily-log-index-fixed-throughput-and-query.json:
```diff
@@ -0,0 +1,137 @@
+  "name": "index-fixed-load-and-query",
+  "description": "Indexes {{ p_number_of_days }} days of logs (each {{ p_daily_logging_volume }} size) running at {{ p_bulk_indexing_throughput }} docs/s while running queries concurrently",
+  "meta": {
+    "client_count": {{ p_bulk_indexing_clients }},
```
Should this be `bulk_client_count` (or something along these lines) as there are bulk-indexing clients and search clients?

danielmitterdorfer commented on README.md:
```diff
@@ -174,8 +177,22 @@ Indexes several days of logs with a fixed (raw) logging volume per day and runni
+| `bulk_indexing_reqs_per_sec` | Number of bulk indexing requests/sec. Multiply this by bulk_size to understand indexing throughput in docs/s.                          | `int` | `20`                  |                                                    
+| `bulk_size`                  | Number of documents to send per bulk                                                                                                   | `int` | `1000`                |
+| `bulk_indexing_clients`      | Number of bulk indexing clients/connections                                                                                            | `int` | `8`                   |
+| `search_clients`             | Number of search clients/connections                                                                                                   | `int` | `1`                   |
```
Maybe we should clarify that each of the three queries will be run with that many clients? I.e. with `search_clients` set to `2`, we will actually run with six clients because there are three searches and each of them is run with two clients.

danielmitterdorfer commented on README.md:
```diff
@@ -174,8 +177,22 @@ Indexes several days of logs with a fixed (raw) logging volume per day and runni
+| `bulk_size`                  | Number of documents to send per bulk                                                                                                   | `int` | `1000`                |
+| `bulk_indexing_clients`      | Number of bulk indexing clients/connections                                                                                            | `int` | `8`                   |
+| `search_clients`             | Number of search clients/connections                                                                                                   | `int` | `1`                   |
+| `daily_logging_volume`       | The raw logging volume. Supported units are bytes (without any unit), `kb`, `MB` and `GB`). For the value, only integers are allowed.  | `str` | `100GB`               |
```
`kb` -> `kB`

danielmitterdorfer commented on eventdata/challenges/daily-log-volume-index-and-query.json:
```diff
@@ -41,10 +41,21 @@
     },
+    {
+      "name": "check-cluster-health",
+      "operation": {
```
we should add a `"retry-until-success": true` here (similar to https://github.com/elastic/rally-eventdata-track/pull/97).

danielmitterdorfer commented on eventdata/challenges/daily-log-volume-index.json:
```diff
@@ -30,12 +30,23 @@
     },
+    {
+      "name": "check-cluster-health",
+      "operation": {
```
we should add a `"retry-until-success": true` here (similar to https://github.com/elastic/rally-eventdata-track/pull/97).

danielmitterdorfer reviewed (commented):
Thanks! Looks mostly fine but I think we need to wait longer in the check-cluster-health calls.

dliappis commented:
@danielmitterdorfer addressed your comments (thanks) in https://github.com/elastic/rally-eventdata-track/pull/99/commits/7d55d84458e23e9533e620d76ac10f1771cbc091.

danielmitterdorfer reviewed (approved):
Thanks! LGTM
